web/server/codechecker_server/permissions.py

```diff
@@ -293,21 +293,51 @@ def __get_perm_record(self, auth_name, is_group):
                 query(SysPerm). \
                 filter(and_(
                     SysPerm.permission == self._perm_name,
-                    SysPerm.name == auth_name,
+                    func.lower(SysPerm.name) == auth_name.lower(),
                     SysPerm.is_group == is_group
                 )).one_or_none()
 
             return record
 
+        def __get_stored_auth_name_and_permissions(self, auth_name, is_group):
+            SysPerm = config_db_model.SystemPermission
+
+            stored_auth_name = auth_name
+            permission_set = set()
+            for name, permission in self.__session.query(
+                    SysPerm.name, SysPerm.permission).filter(and_(
+                    func.lower(SysPerm.name) == auth_name.lower(),
+                    SysPerm.is_group == is_group)):
+                stored_auth_name = name
+                permission_set.add(permission)
+
+            return (stored_auth_name, permission_set)
```

Comment:
This function is very similar to the function which can be found in the `ProductPermission` class. The only difference is the query in the for loop. I think we can use inheritance to avoid code duplication.

Reply:
The implementation can be refactored to eliminate duplication that you mention. And type declarations and docstrings also missing. In  one step these modifications hide the original issue. So it will be soon in an other PR.